### PR TITLE
Add HyperlinkedRelatedField and SerializerMethodHyperlinkedRelatedField fields

### LINKED ISSUE #296: Include only a link in the relationships object
According to the JSONAPI specification, if I have an `Article` model with a to-many relationship to a `Comment` model and I want to express this when I retrieve an `Article` resource, there are multiple valid ways I can do this:

1. Include the comments by default in the response, having a list of resource identifier objects in the `relationships` object and a list with the actual resource objects in the `included` object.
2. Return a list of resource identifier objects in the `relationships` object only, without including the actual attributes of the comments.
3. Include only a link (e.g. `/articles/1/comments`) in the `relationships` object.

I have a model that links to a lot of related objects so the 3rd option would be the best for me, but I can't configure the ResourceRelatedField to act like that. It always returns a `data` object with the full list of related resource identifiers. Is there a way I can avoid this? Or do I have to extend the class myself? Doesn't seem very trivial to do that in a clean way.

Ideally, since the JSONAPI specs accept these 3 options, I think this library should allow these 3 configurations out of the box.

### PR DESCRIPTION
**Pull request by Anton-Shutik**
Fixes #296 

## Description of the Change

Include only links in relationships objects by introducing `HyperlinkedRelatedField` and `SerializerMethodHyperlinkedRelatedField`.

## Checklist

- [x] PR only contains one change (considered splitting up PR)
- [x] unit-test added
- [x] documentation updated
- [x] changelog entry added to `CHANGELOG.md`
- [x] author name in `AUTHORS`

### CODE REVIEW

**Comment by sliverc** [comment]
Thanks for looking into this. This approach is more DRF like so I prefer it over #443. I hoped though that `HyperlinkRelatedField` doesn't execute the query itself. But when I looked at the DRF code it seems it still needs `SkipField` workaround.

I have a feeling that this is actually an issue in DRF itself as `HyperlinkRelatedField` does execute a query for m2m relations even though it doesn't need it. If it is fixed there we could adjust this PR accordingly.

So what I suggest before we move forward is to investigate whether the same problem with executing of queries also exist in DRF `HyperlinkRelatedField` itself. If yes we open an issue in DRF and see what the feedback is. Once we have feedback I think we better now how adjust this PR accordingly.

Would be great if you would have time to follow up on this investigation but also understand when you can't. I am also interested in a fix for this but I am not sure when I will have time to get to it.

**Comment by Anton-Shutik** [comment]
@sliverc I think DRF's `HyperLinkRelatedField` works correct with `many=True`
Let's [see the docs](http://www.django-rest-framework.org/api-guide/relations/#hyperlinkedrelatedfield), `tracks` field:

```
{
    'album_name': 'Graceland',
    'artist': 'Paul Simon',
    'tracks': [
        'http://www.example.com/api/tracks/45/',
        'http://www.example.com/api/tracks/46/',
        'http://www.example.com/api/tracks/47/',
        ...
    ]
}
```
It generates one link per each item in the list, but not one link for related collection, like we do in json-api.
So, I think we have to implement custom logic for this with `ManyRelatedFieldWithNoData`.

Are there other things you're confused with ?

**Comment by sliverc** [comment]
@Anton-Shutik 
This makes total sense. Thanks for the follow up. Let's do it the way you have proposed it then it this PR. Could you follow up with tests, docs CI fixes in this PR?
Once done I would do a more detailed review of the code.

**Comment by sliverc** [comment]
Just on another note why I prefer this to `render_data` flag.

I can simply overwrite `serializer_related_field` like:
```python
class MySerializer(ModelSerializer):
  serializer_related_field = HyperlinkRelatedField
```

this way all related fields are only represented as links without defining each one again in the serializer.

File: docs/usage.md [hunk 1]
```diff
@@ -427,6 +429,12 @@ class LineItemViewSet(viewsets.ModelViewSet):
         return queryset
 ```
 
+#### HyperLinkedRelatedField
+
+In order to improve performance by saving some sql queries we can skip `data`
```
[the remark this answers is not in the document]
Comment (Anton-Shutik) [reply docs/usage.md]: Added

File: rest_framework_json_api/relations.py [hunk 2]
```diff
@@ -140,6 +125,78 @@ def get_links(self, obj=None, lookup_field='pk'):
             return_data.update({'related': related_link})
         return return_data
 
+
+class HyperLinkedRelatedField(HyperLinkedMixin, SkipDataMixin, RelatedField):
```
[the remark this answers is not in the document]
Comment (Anton-Shutik) [reply rest_framework_json_api/relations.py]: Done

**Comment by sliverc** [comment]
Merged. Thanks for your effort. In terms of release date: we do not have a fix release schedule but try to release often but not after every PR.

**Comment by Anton-Shutik** [comment]
@jasonm Enjoy! :)

**Comment by benjaminlong** [comment]
Thanks @Anton-Shutik ! Just integrated your work, it was smooth. Love it !

**Pull request merged**